hs_core/page_processors.py

```diff
@@ -50,14 +52,24 @@ def get_page_context(page, user, resource_edit=False, extended_metadata_layout=N
 
     metadata_status = _get_metadata_status(content_model)
 
-    relevant_tools = []
-    for res_type in ToolResourceType.objects.all():
-        if str(content_model.content_model).lower() in str(res_type.tool_res_type).lower():
-            url = res_type.content_object.url_bases.first()
-            if url:
-                tl = {'title': res_type.content_object.title,
-                      'url': "{}{}{}".format(url.value, "/?res_id=", content_model.short_id)}
-                relevant_tools.append(tl)
+
+    relevant_tools = None
+    if not resource_edit: # view mode
+        relevant_tools = []
+        content_model_str = str(content_model.content_model).lower()
+        for res_type in SupportedResTypes.objects.all():
+            if content_model_str in str(res_type.get_supported_res_types_str()).lower():
+                url_obj = res_type.content_object.url_bases.first()
+                tool_res_obj = hydroshare.get_resource_by_shortkey(url_obj.resShortID)
+                if tool_res_obj:
+                    tool_edit_mode = authorize(request, tool_res_obj.resShortID)[1]
```

Comment:
@selimnairb I just corrected the code. Upon further review, I understand what the parameters to the `authorize()` function do. I now pass both `edit=True` and `raises_exception=False` because an exception should never be raised since this authorization check will only determine if the "Relevant Tools" tab should be added or not. I add the [1] list accessor at the end because `authorize()` returns three values, but only the second one will be collected in a variable.